checkov/common/runners/runner_registry.py

```diff
@@ -99,3 +114,74 @@ def enrich_report_with_guidelines(scan_report, guidelines):
         for record in scan_report.failed_checks + scan_report.passed_checks + scan_report.skipped_checks:
             if record.check_id in guidelines:
                 record.set_guideline(guidelines[record.check_id])
+
+    @staticmethod
+    def get_enriched_resources(report, repo_root):
+        parser = Parser()
+        tf_definitions = {}
+        parsing_errors = {}
+        Parser.parse_directory(
+            parser,
+            directory=repo_root,  # assume plan file is in the repo-root
+            out_definitions=tf_definitions,
+            out_parsing_errors=parsing_errors,
+        )
```

Comment:
you initalize the `Parser` class here and then pass it as a parameter into a method of the class which is not annotated as `@classmethod` please don't do this. Just use directly `parser.parse_directory(directory=...)
```suggestion
        tf_definitions = {}
        parsing_errors = {}
        Parser().parse_directory(
            directory=repo_root,  # assume plan file is in the repo-root
            out_definitions=tf_definitions,
            out_parsing_errors=parsing_errors,
        )
```